### PR TITLE
BASH-1 Config value to display or not the URL on hover at bottom of screen

### PR DESCRIPTION
**Pull request by csduarte**
**Description**
This PR allows configurable url preview (the small gray bar in bottom corner) on mouse hovers over a link.

- [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

**Test Cases**
Load application and hover over any link tag (<a>). See gray bar with url in lower corner.
Disable Config and hover any tag, no gray bar rendered
```json 
//override.js
{
  “1”: {
    "displayUrlOnHover": false
  }
}
```
**Notes**
Includes bash-20

screen shot 2017-09-08 at 10 54 49 am

### CODE REVIEW

File: src/browser/components/MainPage.jsx [hunk 1]
```diff
@@ -315,12 +318,12 @@ const MainPage = createReactClass({
           transitionEnterTimeout={300}
           transitionLeaveTimeout={500}
         >
-          { (this.state.targetURL === '') ?
-            null :
+          { (defaultConfig.displayUrlOnHover === true && this.state.targetURL !== '') ?
```
Comment (yuya-oc) [inline src/browser/components/MainPage.jsx]: `defaultConfig` is explicitly used. So can only maintainer change the setting on source code?

Comment (dmeza) [reply src/browser/components/MainPage.jsx]: @yuya-oc yes for this implementation that's the case. You enable/disable the display of URL hover before building and dist.

**Comment by jasonblais** [comment]
@yuya-oc Sorry, for some reason I couldn't find the auto-generated test build for this PR? Can you help share a link for it?

I'm wondering if we just turn them off in the desktop app by default, without a configurable setting. That seems to the standard in native desktop apps, at least on Windows.

**Comment by yuya-oc** [comment]
@jasonblais https://circleci.com/gh/mattermost/desktop/1007#artifacts And sorry, too. I had to paste the link anyway.

**Comment by jasonblais** [comment]
Hm, I guess we'd need a test build with the following, to properly test this out. 

```
//override.js
{
  “1”: {
    "displayUrlOnHover": false
  }
}
```

@yuya-oc Wondering if this is something you'd be able to help with?

**Comment by yuya-oc** [comment]
@jasonblais These artifacts disabled the hovering URL. https://circleci.com/gh/yuya-oc/desktop/379#artifacts

**Comment by jasonblais** [comment]
@yuya-oc I still seem to see the URLs on hover. I tried downloading the artifact twice,

**Comment by yuya-oc** [comment]
@jasonblais Sorry, I had a mistake at `override.json`. The artifact will appear at https://circleci.com/gh/yuya-oc/desktop/380#artifacts

**Comment by jasonblais** [comment]
Thanks! It seems to work as expected.

@csduarte @dmeza Question: Would you like to have ALL links disappear, or would you like https://github.com or `[Markdown link](some link)` to show on hover?

We have a ticket server-side ([PLT-6828](https://mattermost.atlassian.net/browse/PLT-6828)) to hide some of the URLs on hover, both on browser and the desktop app. This includes the channel names in the sidebar for instance. Was just wondering if this would meet your requirements, or if you'd still want none of the links to display the URL on hover.

**Comment by csduarte** [comment]
@jasonblais I think the differentiation should be: 
- UI links hover info should be hidden, the link content should be obvious based on the UI itself
- User generated links should be `hoverable` since the link content may not obvious

**Comment by jasonblais** [comment]
@csduarte, yes, that's what I had in mind but you summarized it much better than I did :) 

If UI links hover info is hidden through https://mattermost.atlassian.net/browse/PLT-6828, will you need this setting in that case?

**Comment by jasonblais** [comment]
PS: Here is a PR which removes UI links hover info. It'll apply to both the webapp and desktop app starting in Mattermost serverv 4.3: https://github.com/mattermost/mattermost-webapp/pull/53

It'll keep the link hovers for user generated links.

**Comment by jasonblais** [comment]
The server PR is now under dev review, which hides internal links on hover: https://github.com/mattermost/mattermost-webapp/pull/53

The only links it doesn't hide are when hovering over teams and channels in the left-hand side on the desktop app. These will be handled by https://mattermost.atlassian.net/browse/PLT-7773

Hence, adding the `Pending` label, given this PR may not be needed once the server-side changes are merged.

**Comment by jasonblais** [comment]
The server PR is now under dev review, which hides internal links on hover: https://github.com/mattermost/mattermost-webapp/pull/53

The only links it doesn't hide are when hovering over teams and channels in the left-hand side on the desktop app. These will be handled by https://mattermost.atlassian.net/browse/PLT-7773

Hence, adding the `Pending` label, given this PR may not be needed once the server-side changes are merged.

/cc @yuya-oc @csduarte @dmeza

**Comment by jasonblais** [comment]
Server-side changes are merged and will be part of the Mattermost server v4.4 release on the 16th.

Hence, closing this PR as it no longer applies.

@csduarte @dmeza Let me know if any updates are needed for this, or if the server-side changes don't cover your case.

**Pull request closed without merging**